### PR TITLE
Added data test id's

### PR DESCRIPTION
**Pull request by CalebGerman**

### CODE REVIEW

File: packages/changed-elements-react/src/widgets/ChangedElementsWidget.tsx [hunk 1]
```diff
@@ -302,7 +302,7 @@ export class ChangedElementsWidget extends Component<ChangedElementsWidgetProps,
         </IconButton>
         {
           this.props.useV2Widget &&
-          <InfoButton title={IModelApp.localization.getLocalizedString("VersionCompare:versionCompare.versionCompare")} message={this._widgetInfo} />
+          <InfoButton dataTestId="⁠comparison-legend-widget-info" title={IModelApp.localization.getLocalizedString("VersionCompare:versionCompare.versionCompare")} message={this._widgetInfo} />
```
Comment (roluk) [inline packages/changed-elements-react/src/widgets/ChangedElementsWidget.tsx:305]: Why do buttons need data-testid? Can't someone just match the button by role=button and the button label?

Comment (roluk) [reply packages/changed-elements-react/src/widgets/ChangedElementsWidget.tsx:305]: Oh, these buttons don't have user-readable labels. That's fine then.

File: packages/changed-elements-react/src/widgets/ChangedElementsWidget.tsx [hunk 2]
```diff
@@ -380,7 +381,7 @@ export class ChangedElementsWidget extends Component<ChangedElementsWidgetProps,
             {this.state.loaded ? this.getChangedElementsContent() : this.getLoadingContent()}
           </WidgetComponent.Body>
           <WidgetComponent.ToolBar>
-            {(this.props.useV2Widget && (!!this.props.feedbackUrl)) && <FeedbackButton feedbackUrl={this.props.feedbackUrl ?? ""}></FeedbackButton>}
+            {(this.props.useV2Widget && (!!this.props.feedbackUrl)) && <FeedbackButton data-testId="⁠comparison-widget-v2-feedback-btn" feedbackUrl={this.props.feedbackUrl ?? ""}></FeedbackButton>}
```
Comment (diegopinate) [inline packages/changed-elements-react/src/widgets/ChangedElementsWidget.tsx:384]: Does casing matter?
```suggestion
            {(this.props.useV2Widget && (!!this.props.feedbackUrl)) && <FeedbackButton data-testid="⁠comparison-widget-v2-feedback-btn" feedbackUrl={this.props.feedbackUrl ?? ""}></FeedbackButton>}
```

File: packages/changed-elements-react/src/widgets/ChangedElementsWidget.tsx [hunk 3]
```diff
@@ -396,6 +397,7 @@ export class ChangedElementsWidget extends Component<ChangedElementsWidgetProps,
           <V2DialogProvider>
             {this.state.versionSelectDialogVisible &&
               <VersionCompareSelectDialogV2
+                data-testId="⁠comparison-widget-v2-modal"
```
Comment (diegopinate) [inline packages/changed-elements-react/src/widgets/ChangedElementsWidget.tsx:400]: Does casing matter?
```suggestion
                data-testid="⁠comparison-widget-v2-modal"
```

File: packages/changed-elements-react/src/widgets/FeedbackButton.tsx [hunk 4]
```diff
@@ -22,6 +23,7 @@ export function FeedbackButton(props: Props) {
       href={props.feedbackUrl}
       target="_blank"
       rel="noreferrer"
+      data-testid={props["data-testId"]}
```
Comment (diegopinate) [inline packages/changed-elements-react/src/widgets/FeedbackButton.tsx:26]:
```suggestion
      data-testid={props["data-testid"]}
```

File: packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx [hunk 5]
```diff
@@ -25,6 +25,7 @@ export interface VersionCompareSelectDialogV2Props {
   iModelConnection: IModelConnection;
   /** onClose triggered when user clicks start comparison or closes dialog.*/
   onClose: (() => void) | undefined;
+  "data-testId"?: string;
```
Comment (diegopinate) [inline packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx:28]:
```suggestion
  "data-testid"?: string;
```

File: packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx [hunk 6]
```diff
@@ -130,6 +131,7 @@ export function VersionCompareSelectDialogV2(props: VersionCompareSelectDialogV2
   };
   return (
     <Modal
+      data-testid={props["data-testId"]}
```
Comment (diegopinate) [inline packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx:134]:
```suggestion
      data-testid={props["data-testid"]}
```

**Pull request merged**